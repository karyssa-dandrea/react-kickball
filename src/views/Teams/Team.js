import React from 'react';
import TeamDetail from '../../components/TeamFunction/TeamDetail';
import { fetchTeamById } from '../../services/teams';
import { useEffect, useState } from 'react';

export default function Team(props) {
  const [team, setTeam] = useState([]);
  const id = props.match.params.id;
  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchTeamById(id);
      setTeam(data);
    };
    fetchData();
  }, [id]);
  return (
    <div className="team-deets">
      <p>{team.name}</p>
      <p>{team.city}</p>
      <p>{team.state}</p>
      <TeamDetail team={team}></TeamDetail>
    </div>
  );
}

// make and add a teamdetail component
// write function for getteambyId
// pass an id on line 10 into that function
//7 1/2 define a const id using params.match
