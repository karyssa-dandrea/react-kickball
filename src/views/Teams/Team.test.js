import { render, screen } from '@testing-library/react';
import Team from '../../views/Teams/Team';
import { MemoryRouter } from 'react-router-dom';

test('should render team details', async () => {
  const { container } = render(
    <MemoryRouter>
      <Team match={{ params: { id: 2 } }} />
    </MemoryRouter>
  );
  await screen.findByText('Stumptown Lumberjacks');
  expect(container).toMatchSnapshot();
});
