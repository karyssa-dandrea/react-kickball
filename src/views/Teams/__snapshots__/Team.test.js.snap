// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render team details 1`] = `
<div>
  <div
    class="team-deets"
  >
    <p>
      Stumptown Lumberjacks
    </p>
    <p>
      Portland
    </p>
    <p>
      OR
    </p>
    <div
      class="team-details"
    >
      <a
        href="/players/5"
      >
        Gary Coolman
      </a>
      <a
        href="/players/6"
      >
        Terry Cruise
      </a>
      <a
        href="/players/7"
      >
        Bud E. Guy
      </a>
      <a
        href="/players/8"
      >
        Buster Rimes
      </a>
    </div>
  </div>
</div>
`;
